enhancements/windows-containers/container-runtime-containerd.md

```diff
@@ -0,0 +1,205 @@
+status: implementable
+---
+
+# containerd - new container runtime  
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [x] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [OpenShift-docs](https://github.com/OpenShift/OpenShift-docs/)
+
+## Summary
+
+This enhancement intends to allow customers to bring up Windows node with containerd
+as default runtime from OpenShift 4.11 onwards. When customers upgrade clusters to
+OpenShift 4.11, CRI runtime will be migrated from Docker to containerd
+
+## Motivation
+
+In Kubernetes, the CRI interface has been used to talk to a container runtime. From
+Kubernetes 1.24 onwards dockershim will be removed from kubelet code. At present
+WMCO uses Docker as the default runtime. The goal is to make containerd as default
+runtime and move away from Docker before dokershim has been decoupled from kubelet.
+
+## Goals
+
+As part of this enhancement we plan to do the following:
+* Make containerd the default CRI runtime on Windows nodes.
+* When WMCO is upgraded from 5.x.z to 6.x.z, the runtime on all existing Windows nodes will be migrated to
+  containerd.  
+
+### Non-Goals
+
+* De-configuring Docker runtime is not part of this enhancement.
+* Allowing the user to choose a  container runtime migrating back to Docker from containerd is not supported.
+*  Downgrade is not supported.
+
+## Proposal
+
+To make containerd the default runtime, containerd should be installed as a Windows
+service before kubelet gets installed. While starting kubelet, parameters `container-runtime` and
+`container-runtime-endpoint`should be configured. The remaining WMCO workflow has no impact other than the proposed
+changes. MachineSet and BYOH (Bring Your Own Host) upgrade details are discussed in the upgrade section.
+There is no difference between MachineSet and BYOH on how we enable containerd as the default runtime.
+
+Golden Image:
+
+In MachineSet use case, the Windows image has preinstalled Docker runtime. Users can create a new Windows
+image excluding Docker runtime or continue to use the same Windows image. Both images will be supported.
+
+Containerd Migration plan:
+
+Containerd will be the default runtime in OpenShift 4.11. For development, we plan to introduce a
+WMCO CLI option which allows us to choose the runtime. This CLI option will be used in the
+release of the WMCO community operator for OpenShift 4.10 to allow customers to try out Windows
+nodes with containerd. Customers can use 4.10 community WMCO to create new clusters or upgrade
+existing clusters.
+   
+### User Stories
+
+Stories can be found within the [Windows Containers: containerd epic](https://issues.redhat.com/browse/WINC-505)
+
+### Justification
+We have explored containerd, CRIO, and Mirantis dockershim as an alternate runtime. After looking into business and
```

Comment:
I have completely rewritten the paragraph below and statement address your comment in `Disadvantage of using other runtimes` section.
>Mirantis supported dockershim will continue to use Docker. Using Docker as runtime has an impact on memory
and CPU performance.

Let me know if you think otherwise.